src/reducers/7.12/reducer.ts

```diff
@@ -1698,23 +1710,63 @@ export const getConnector = (cr: BrokerCR, name: string) => {
   return undefined;
 };
 
+export const getAllConnectors = (brokerModel: BrokerCR) => {
+  return brokerModel.spec?.connectors || [];
+};
+
+export const generatePortNumber = (usedPorts: number[]): number => {
+  let port = 5555;
+  while (usedPorts.includes(port)) {
+    port += 1;
+    //console.log('increament port', port);
+  }
+  return port;
+};
+
 export const getConfigPort = (
   brokerModel: BrokerCR,
   configType: ConfigType,
   configName: string,
 ): number => {
-  if (configType === ConfigType.connectors) {
-    const connector = getConnector(brokerModel, configName);
-    if (connector?.port) {
-      return connector.port;
-    }
-  }
   if (configType === ConfigType.acceptors) {
     const acceptor = getAcceptor(brokerModel, configName);
     if (acceptor?.port) {
+      //console.log(`Existing port for acceptor ${configName}: ${acceptor.port}`);
       return acceptor.port;
     }
+
+    const acceptors = getAllAcceptors(brokerModel);
+    const usedPorts: number[] = [];
+    acceptors.forEach((acceptor) => {
+      if (acceptor.port) {
+        usedPorts.push(acceptor.port);
+      }
+    });
+    const newPort = generatePortNumber(usedPorts);
+    //console.log(`Generated new port for acceptor ${configName}: ${newPort}`);
+    return newPort;
+  } else if (configType === ConfigType.connectors) {
+    const connector = getConnector(brokerModel, configName);
+    if (connector?.port) {
+      // console.log(
+      //   `Existing port for connector ${configName}: ${connector.port}`,
+      // );
+      return connector.port;
+    }
+
+    const connectors = getAllConnectors(brokerModel);
+    const usedPorts: number[] = [];
+    connectors.forEach((connector) => {
+      if (connector.port) {
+        usedPorts.push(connector.port);
+      }
+    });
+    const newPort = generatePortNumber(usedPorts);
+    //console.log(`Generated new port for connector ${configName}: ${newPort}`);
+    return newPort;
```

Comment:
The function `getConfigPort` is a getter function, it's purpose it to give an easy access to some data we care about. It should not be used to generate a port number.

Can you explain me what was your plan for this section of the code? So that we can find a better way to integrate it.